zorder/src/main/scala/org/locationtech/sfcurve/zorder/Z2.scala

```diff
@@ -97,54 +100,63 @@ object Z2 {
     *
     * @return (common prefix, number of bits in common)
     */
-  def longestCommonPrefix(lower: Long, upper: Long): (Long, Int) = {
-    var bitShift = MAX_BITS - MAX_DIM
+  def longestCommonPrefix(lower: Long, upper: Long): ZPrefix = {
+    var bitShift = TOTAL_BITS - MAX_DIM
     while ((lower >>> bitShift) == (upper >>> bitShift) && bitShift > -1) {
       bitShift -= MAX_DIM
     }
     bitShift += MAX_DIM // increment back to the last valid value
-    (lower & (Long.MaxValue << bitShift), MAX_BITS - bitShift)
+    ZPrefix(lower & (Long.MaxValue << bitShift), 64 - bitShift)
   }
 
-  /** Recurse down the quad-tree and report all z-ranges which are contained in the rectangle defined by the min and max points */
-  def zranges(min: Z2, max: Z2, globalMaxRecurse: Int = 32): Seq[IndexRange] = {
-    val (commonPrefix, commonBits) = longestCommonPrefix(min.z, max.z)
+  /**
+    * Recurse down the quad-tree and report all z-ranges which are contained
+    * in the cube defined by the min and max points
+    *
+    * @param min lower bound
+    * @param max upper bound
+    * @param precision bit precision of the z-values. can be used to stop searching after
+    *                  considering a certain number of bits.
+    * @return
+    */
+  def zranges(min: Z2, max: Z2, precision: Int = 64): Seq[IndexRange] = {
```

Comment:
It feels as if this 64, and the one in ```longestCommonPrefix```, ought to be constants, but perhaps that's just me being pedantic.